lib/sycamore/sycamore/connectors/elasticsearch/elasticsearch_writer.py

```diff
@@ -0,0 +1,140 @@
+from dataclasses import dataclass, field
+from typing import Any, Optional
+from sycamore.connectors.common import drop_types, flatten_data
+from typing_extensions import TypeGuard
+
+from sycamore.data.document import Document
+from sycamore.connectors.base import BaseDBWriter
+
+from elasticsearch import Elasticsearch, ApiError
+from elasticsearch.helpers import parallel_bulk
+
+
+@dataclass
+class ElasticClientParams(BaseDBWriter.ClientParams):
+    url: str
+    es_client_args: dict = field(default_factory=lambda: {})
+
+
+@dataclass
+class ElasticTargetParams(BaseDBWriter.TargetParams):
+    index_name: str
+    mappings: dict[str, Any] = field(
+        default_factory=lambda: {
+            "properties": {
+                "embeddings": {
+                    "type": "dense_vector",
+                    "dims": 384,
+                    "index": True,
+                    "similarity": "cosine",
+                },
+                "properties": {"type": "object"},
+            }
+        }
+    )
+    flatten_properties: bool = False
+
+    def compatible_with(self, other: BaseDBWriter.TargetParams) -> bool:
+        if not isinstance(other, ElasticTargetParams):
+            return False
+        if self.index_name != other.index_name:
+            return False
+        if self.mappings != other.mappings:
+            return False
```

Comment:
I feel like this is gonna bite you when you try to write to an existing index and there are already mappings for already-existing documents.

Reply:
Fair, have added a better way of checking this